src/atoms/atomgraph.jl

```diff
@@ -139,49 +94,44 @@ function AtomGraph(
 )
 
     local ag
-    local to_build_graph = true
-    to_serialize = !isnothing(output_file_path)
 
-    if to_serialize
-        if isfile(output_file_path)
-            if overwrite_file
-                @info "Output file already exists and `overwrite_file` is set to false; returning deserialized AtomGraph at $output_file_path. If you wanted to rebuild the graph, set `overwrite=true`."
-                to_build_graph = false
-                ag = deserialize(output_file_path)
-            end
-        end
+    if !isfile(input_file_path)
+        @warn "$input_file_path does not exist. Cannot build graph from a non-existent file."
+        return missing
     end
 
-    if to_build_graph
-        if splitext(input_file_path)[end] == ".jls"
-            ag = deserialize(input_file_path)
+    if splitext(input_file_path)[end] == ".jls" # deserialize
+        ag = deserialize(input_file_path)
+        if isempty(id)   # if an id is specified, use that instead of whatever id is got from deserializing
```

Comment:
wait then shouldn't this be `if !isempty(id)`?

Reply:
Oh yes, it should. Sorry about that. 😅

Reply:
🥁 more tests more tests

Reply:
Resolved in 7257df3. (for now)
We won't need this after all, as the `id` is set to `input_file_path` now.
If someone wants to intentionally have an empty string in there, I say let them. 😆